joern-cli/frontends/php2cpg/src/main/scala/io/joern/php2cpg/astcreation/AstCreatorHelper.scala

```diff
@@ -155,16 +159,60 @@ trait AstCreatorHelper(disableFileContent: Boolean)(implicit withSchemaValidatio
         }
 
         scope.addToScope(name, local) match {
-          case BlockScope(block, _)           => diffGraph.addEdge(block, local, EdgeTypes.AST)
-          case MethodScope(_, block, _, _, _) => diffGraph.addEdge(block, local, EdgeTypes.AST)
-          case _                              => // do nothing
+          case BlockScope(block, _)              => diffGraph.addEdge(block, local, EdgeTypes.AST)
+          case MethodScope(_, block, _, _, _, _) => diffGraph.addEdge(block, local, EdgeTypes.AST)
+          case _                                 => // do nothing
         }
 
         local
+      case Some(local: NewLocal) if scope.isSurroundedByArrowClosure && local.closureBindingId.isDefined => local
+      case Some(param: NewMethodParameterIn)
+          if scope.isSurroundedByArrowClosure && !scope.surroundingMethodParams.contains(param.name) =>
+        val methodName = scope.surroundingScopeFullName.get // should throw error if this does not exist
+        val methodRef  = scope.getSurroundingArrowClosureMethodRef
+
+        val closureBindingId = s"$methodName:$name"
+        val localNode_ =
+          localNode(expr, name, param.code, Defines.Any, closureBindingId = Option(closureBindingId))
+
+        createClosureBinding(methodRef, closureBindingId, param, localNode_)
+
+        localNode_
+      case Some(local: NewLocal) if scope.isSurroundedByArrowClosure =>
+        val methodName = scope.surroundingScopeFullName.get // should throw error if this does not exist
+        val methodRef  = scope.getSurroundingArrowClosureMethodRef
+
+        val closureBindingId = s"$methodName:$name"
+        val localNode_ =
+          localNode(expr, name, local.code, local.typeFullName, closureBindingId = Option(closureBindingId))
+
+        createClosureBinding(methodRef, closureBindingId, local, localNode_)
+
+        localNode_
       case Some(local) => local
     }
   }
 
+  private def createClosureBinding(
+    methodRef: Option[NewMethodRef],
+    closureBindingId: String,
+    existingNode: (NewLocal | NewMethodParameterIn),
+    newLocalNode: NewLocal
+  ): Unit = {
+    val closureBindingNode =
+      NewClosureBinding().closureBindingId(closureBindingId).evaluationStrategy(EvaluationStrategies.BY_SHARING)
```

Comment:
There already two other places in the php2cpg frontend which create a closure binding from scratch ( `NewClosureBinding()`. Is it possible to get some sensible code reuse with those?